### PR TITLE
Migration to Standalone Components

### PR DESCRIPTION
**Pull request by ssaring**
Context: CFWEB-68 Incorporate Standalone Components into Styleguide

Example app has been migrated partially to Standalone Components / API. Changes: 
* Main app root components is standalone now.
* App is bootstrapped by the standalone API, no AppModule and AppRouteModule anymore.
* Package 'shared' is completely migrated to the standalone approach.
* Package 'order' is completely migrated to the standalone approach.

Package 'catalog' keeps the module-based approach, as an example for integration of the module-based and standalone-based packages inside an app.

### CODE REVIEW

**Review by Azaraky — APPROVED** [review]

File: src/app/app.config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Carl Zeiss AG
+ * SPDX-License-Identifier: MIT
+ */
+
+import { ApplicationConfig, importProvidersFrom } from '@angular/core';
+import { provideRouter } from '@angular/router';
+
+import { routes } from './app.routes';
```
Comment (manuel-mauky) [inline src/app/app.config.ts:9]: In some places imports are done with `@app/<something>` and sometimes with `./<something>`.

Shouldn't we try to have a common style here?

Maybe this is a topic for a different PR and I'm not sure what's best practice in realworld apps.

**Comment by manuel-mauky** [comment]
The PR looks good. 
My only issue is that the idea that the catalog package keeps the old module approach isn't documented yet. Can we document this somewhere? 
First, it would help people who are looking for exactly this combination.
And second, we ourselves would know in the future that the catalog package is by intension in the old format and we should keep it like that.

File: src/app/app.config.ts (hunk 1 above)
Comment (ssaring) [reply src/app/app.config.ts:9]: My understanding for the imports path handling is (and the tooling seems to work the same way):
* when the imported file is in the same directory, use "from './foo.bar'". 
* when importing from other directories, try to use the defined @ shortcuts

Most of the project files are importing this way, despite a few exceptions (not part of this PR). Therefore I would suggest to address this in a separate branch, if needed. Feel free to reopen this PR issue if you see it differently.

**Comment by ssaring** [comment]
> The PR looks good. My only issue is that the idea that the catalog package keeps the old module approach isn't documented yet. Can we document this somewhere? First, it would help people who are looking for exactly this combination. And second, we ourselves would know in the future that the catalog package is by intension in the old format and we should keep it like that.

@manuel-mauky Good point, added some notes on that in the README.md. Will add further details in the Styleguide documentation later on. Sufficient from your perspective?

**Review by sbley — APPROVED** [review]

**Pull request merged**